pandas/io/common.py

```diff
@@ -324,9 +325,51 @@ def _get_handle(path, mode, encoding=None, compression=None):
         else:
             f = open(path, mode)
 
+    if memory_map and hasattr(f, 'fileno'):
+        try:
+            f = MMapWrapper(f)
+        except:  # fallback: leave file handler as is
```

Comment:
It's really whatever can be thrown by the mmap constructor and while I could certainly guess what some of them might be, not really sure how safe that is.